File: 2021/3/part_2.py
with open("input.txt") as f:
    lines = [l.rstrip("\n") for l in f.readlines() if l != "\n"]

oxy = None
co2 = None

og_lines = lines.copy()
ls_lines = lines.copy()

for i in range(len(lines[0])):
    ones = 0
    zeros = 0

    for l in og_lines:
        if l[i] == "0":
            zeros += 1
        else:
            ones += 1
        
    mcb = "1" if ones >= zeros else "0"

    print("oglines:", og_lines, "mcb:", mcb)
    n_lines = []
    for l in og_lines:
        print(l)
        print(" "*i + "^")
        if l[i] == mcb:
            n_lines.append(l)
    og_lines = n_lines
    if not oxy and len(og_lines) == 1:
        oxy = eval(f"0b{og_lines[0]}")

    ones = 0
    zeros = 0

    for l in ls_lines:
        if l[i] == "0":
            zeros += 1
        else:
            ones += 1
        
    lcb = "1" if zeros>ones else "0"

    n_lines = []
    for l in ls_lines:
        if l[i] == lcb:
            n_lines.append(l)
    ls_lines = n_lines
    if not co2 and len(ls_lines) == 1:
        co2 = eval(f"0b{ls_lines[0]}")

print(oxy,co2)
print(oxy*co2)

File: 2021/3/part_2.js
const fs = require("fs");

const data = fs.readFileSync("input.txt", "utf8");

const lines = data.trim("\n").split("\r").join("").split("\n");

let oxy = null;
let co2 = null;

let ogLines = [...lines];
let lsLines = [...lines];

for (i = 0; i < lines[0].length; i++) {
    let ones = 0;
    let zeros = 0;

    ogLines.forEach(l => {
        if (l[i] === "0") zeros += 1;
        else ones += 1;
    });

    let mcb = ones >= zeros ? "1" : "0";

    ogLines = ogLines.filter(l => l[i] === mcb);

    if (!oxy && ogLines.length === 1) oxy = parseInt(ogLines[0], 2);

    ones = 0;
    zeros = 0;

    lsLines.forEach(l => {
        if (l[i] === "0") zeros += 1;
        else ones += 1;
    });

    lcb = zeros > ones ? "1" : "0";

    lsLines = lsLines.filter(l => l[i] === lcb);

    if (!co2 && lsLines.length === 1) co2 = parseInt(lsLines[0], 2);
}

console.log(oxy * co2);
